controller/iteration_blackbox_test.go

```diff
@@ -1269,4 +1269,73 @@ func (rest *TestIterationREST) TestUpdateIteration() {
 		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
 		test.UpdateIterationBadRequest(t, svc.Context, svc, ctrl, itr1.ID.String(), &payload)
 	})
+
+	rest.T().Run("update fail - parent UUID is same as subject iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(2))
+		typeIterationString := iteration.APIStringTypeIteration
+		itr1 := fxt.Iterations[1]
+		newParentIDStr := itr1.ID.String()
+		payload := app.UpdateIterationPayload{
+			Data: &app.Iteration{
```

Comment:
Do you really need two iterations? I guess one is enough, isn't it? Can't you then just write this in order to create the proper update payload?

```go
payload := app.UpdateIterationPayload{
  Data: ConvertIteration(&http.Request{Host: "api.service.domain.org"}, iteration.Iteration{
    ID: fxt.Iterations[0].ID,
    Path: path.Path{fxt.Iterations[0], fxt.Iterations[0].ID},
  }),
}
```

Reply:
yes, 2 iterations are not needed, removed now.
also, I would prefer to use minimum payload to patch calls.
hence done in our traditional style in https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/75622623f469327dd41526bd841df5dae623e6a2